contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_transform_resize.py

```diff
@@ -0,0 +1,253 @@
+        codec: str,
+        width_from: int,
+        height_from: int,
+        width_to: int,
+        height_to: int,
+        framerate: int,
+    ):
+        self._platform = platform
+        self._codec = codec
+        self._width_from = width_from
+        self._height_from = height_from
+        self._width_to = width_to
+        self._height_to = height_to
+        self._framerate = framerate
+        self._codec_parser_map = {
+            GStreamerEncodePlugins.V4L2H264ENC.value: "h264parse"
+        }
+        # This sample video file will be consumed by any gstreamer piple as
+        # input video.
+        self._golden_sample = get_big_bug_bunny_golden_sample(
+            width=self._width_from,
+            height=self._height_from,
+            framerate=self._framerate,
+        )
+        self._artifact_file = ""
+
+    @property
+    def artifact_file(self) -> str:
+        if not self._artifact_file:
+            self._artifact_file = generate_artifact_name(extension="mp4")
+        return self._artifact_file
+
+    @property
+    def psnr_reference_file(self) -> str:
+        """
+        A golden reference which has been transformed in advance. It's used to
+        be the compared reference file for PSNR.
+        """
+        golden_reference = get_big_bug_bunny_golden_sample(
+            self._width_to, self._height_to, self._framerate
+        )
+
+        full_path = os.path.join(
+            VIDEO_CODEC_TESTING_DATA, SAMPLE_2_FOLDER, golden_reference
+        )
+        if not os.path.exists(full_path):
+            raise SystemExit(
+                "Error: Golden PSNR reference '{}' doesn't exist".format(
+                    full_path
+                )
+            )
+
+        return full_path
+
+    def build_pipeline(self) -> str:
+        """
+        Build the GStreamer commands based on the platform and codec.
+        Returns:
+            str: A GStreamer command based on the platform and
+            codec.
+        """
+        pipeline = (
+            "{} filesrc location={} ! decodebin ! v4l2convert ! "
+            "video/x-raw,width={},height={} ! {} ! {} ! mp4mux ! filesink"
+            " location={}"
+        ).format(
+            GST_LAUNCH_BIN,
+            self._golden_sample,
+            self._width_to,
+            self._height_to,
+            self._codec,
+            self._codec_parser_map.get(self._codec),
+            self.artifact_file,
+        )
+        return pipeline
+
+
+def main() -> None:
+    args = register_arguments()
+    p = project_factory(args)
+    logging.info("Step 1: Generating artifact...")
+    cmd = p.build_pipeline()
+    # execute command
+    execute_command(cmd=cmd)
+    logging.info("\nStep 2: Checking metadata...")
+    mv = MetadataValidator(file_path=p.artifact_file)
+    mv.validate("width", args.width_to).validate(
+        "height", args.height_to
+    ).validate("frame_rate", args.framerate).validate(
+        "codec", args.encoder_plugin
```

Comment:
Are we not validating height anymore? 

(also `<MetadataValidator>.validate` wants an `int` but width_to is `str`. Not really an issue here)